torchdata/datapipes/iter/util/tararchiveloader.py

```diff
@@ -57,9 +57,18 @@ def __iter__(self) -> Iterator[Tuple[str, BufferedIOBase]]:
             validate_pathname_binary_tuple(data)
             pathname, data_stream = data
             try:
-                reading_mode = self.mode if data_stream.seekable() else self.mode.replace(":", "|")
-                # typing.cast is used here to silence mypy's type checker
-                tar = tarfile.open(fileobj=cast(Optional[IO[bytes]], data_stream), mode=reading_mode)
+                if isinstance(data_stream, tarfile.TarFile):
+                    tar = data_stream
```

Comment:
super nit: can we add a small test for this case? (only if it is easy to add)

Reply:
Make sense. I will add one via `map`.

Reply:
After digging deeper on the implementation, I will remove this condition as `validate_pathname_binary_tuple ` won't allow the stream being `tarfile.TarFile` directly. It must be wrapped by `StreamWrapper`.
https://github.com/pytorch/data/blob/6b8c5780d43d80854f340df60825a7e624c1ec81/torchdata/datapipes/iter/util/tararchiveloader.py#L57